### PR TITLE
Redirect command Stdout & Stderr for command_runner

### PR DESCRIPTION
**Pull request by lichen2013**
CombinedOutput waits command complete. When we run a long run command, such as continuously get new log entries, it failed to get run results.

Fixes: #2447

### CODE REVIEW

**Comment by dlorenc** [comment]
@minikube-bot ok to test

File: pkg/minikube/bootstrapper/exec_runner.go [hunk 1]
```diff
@@ -48,11 +48,26 @@ func (*ExecRunner) Run(cmd string) error {
 func (*ExecRunner) CombinedOutput(cmd string) (string, error) {
 	glog.Infoln("Run with output:", cmd)
 	c := exec.Command("/bin/bash", "-c", cmd)
-	out, err := c.CombinedOutput()
+	out, err := c.StdoutPipe()
+	defer out.Close()
```
Comment (dlorenc) [inline pkg/minikube/bootstrapper/exec_runner.go]: CombinedOutput also gets us Stderr, right? I think if you add that this should be fine.

Comment (lichen2013) [reply pkg/minikube/bootstrapper/exec_runner.go]: Updated with another approach, and redirected both Stdout & Stderr

**Comment by lichen2013** [comment]
Did a little bit more check, I have a guess that issue #2447 might not only exists for exec_runner. 
Maybe a better fix should be update the CommandRunner interface ?

File: pkg/minikube/bootstrapper/exec_runner.go [hunk 2]
```diff
@@ -48,11 +48,14 @@ func (*ExecRunner) Run(cmd string) error {
...
+	c.Stdout = os.Stdout
+	c.Stderr = os.Stdout
+	err := c.Run()
 	if err != nil {
-		return "", errors.Wrapf(err, "running command: %s\n output: %s", cmd, out)
+		return "", errors.Wrapf(err, "running command: %s\n.", cmd)
 	}
-	return string(out), nil
+
+	return "", nil
```
Comment (dlorenc) [inline pkg/minikube/bootstrapper/exec_runner.go]: This function is meant to return the combined output. That's the string parameter in the output.

I think we need to do something like you had before, where we capture both stdout and stderr as variables, then run the command and return the combined strings.

Comment (lichen2013) [reply pkg/minikube/bootstrapper/exec_runner.go]: For a long running command, like continues printing new logs, it is not possible to get the return strings. 

I would like to propose the change like this:

```
func (*ExecRunner) CombinedOutputTo(cmd string, out io.Writer) error {
        glog.Infoln("Run with output:", cmd)
        c := exec.Command("/bin/bash", "-c", cmd)
        c.Stdout = out
        c.Stderr = out
        err := c.Run()
        if err != nil {
                return errors.Wrapf(err, "running command: %s\n.", cmd)
        }
        return nil
}
```

```
func (*ExecRunner) CombinedOutput(cmd string) (string, error) {
        var b bytes.Buffer
        err := CombinedOutputTo(cmd,  &b)
        if err != nil {
		return "", errors.Wrapf(err, "running command: %s\n output: %s", cmd, b.Bytes())
        return b.Bytes(), nil
}
```

For command like `minikube logs -f`, it should call `CombinedOutputTo` directly:
```
        CombinedOutputTo(cmd, os.Stdout)
```

**Comment by r2d4** [comment]
I actually think the issue is with the SSHRunner implementation rather than the ExecRunner. Both implement the CommandRunner interface. Neither support streaming.  I agree the interface should probably be changed to accommodate it.

**Comment by lichen2013** [comment]
Tested with vm-driver `none` & `virtualbox`.

**Comment by dlorenc** [comment]
@minikube-bot test this please

**Comment by dlorenc** [comment]
D'oh, looks like a small build error still: pkg/minikube/bootstrapper/localkube/localkube_test.go:209:15: l.GetClusterLogs undefined (type LocalkubeBootstrapper has no field or method GetClusterLogs)

**Comment by dlorenc** [comment]
@minikube-bot test this please

**Pull request merged**